drivers/flash/flash_stm32.c

```diff
@@ -57,6 +57,10 @@ LOG_MODULE_REGISTER(flash_stm32, CONFIG_FLASH_LOG_LEVEL);
 
 #define CFG_HW_FLASH_SEMID	2
 
+static const struct flash_parameters flash_stm32_parameters = {
+	.erase_value = 0xff,
```

Comment:
This doesn't apply to all series.
On a handfull, this should be 0x00.
I'll get back to you with the references

Reply:
OK, thanks for looking into this.

Reply:
@utzig has mentioned same thing here: https://github.com/zephyrproject-rtos/zephyr/pull/25947#discussion_r438000040 though I do not know how does it apply to our stm32 drivers.

Reply:
@erwango Can you advise me on the stm32 characteristic values for `erase_value`?

Reply:
@de-nordic After check, only L0/L1 use 0x00 as erase value (I guess reason is they have an EEPROM and this is easier to use same erase value for both).
Since L0/L1 have no flash driver yet, proposed code is fine. Maybe adding a warning comment in case of L0/L1 future addition would be a nice thing though.

Reply:
@erwango Thanks for the information. Will add warning information.

Reply:
There is a PR for an L0 driver (https://github.com/zephyrproject-rtos/zephyr/pull/25245) that will hit this issue. The easiest way to distinguish between the EEPROM and FLASH implementations is probably to check for the FLASH_CR_LOCK (erase value 0xff) or FLASH_PECR_PELOCK (erase value 0x00). It's not exactly pretty, but it avoids hard-coding the SOC family.

Reply:
> There is a PR for an L0 driver (#25245) that will hit this issue. The easiest way to distinguish between the EEPROM and FLASH implementations is probably to check for the FLASH_CR_LOCK (erase value 0xff) or FLASH_PECR_PELOCK (erase value 0x00). It's not exactly pretty, but it avoids hard-coding the SOC family.

@andysan, does this mean that also for L0 the flash erase value is 0xff?